### PR TITLE
Improve uninstalling NGINX capabilities

### PR DESCRIPTION
**Pull request by alessfg**
### Proposed changes

BREAKING CHANGES:

* The `nginx_state` variable has been replaced with `nginx_setup` and instead of using `present`, `absent`, `latest` you should now use `install`, `uninstall` and `update`.
* `nginx_install` variable is no more. Use `nginx_enable` instead.

BUG FIXES:

Uninstalling NGINX should now work correctly under most scenarios.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/ansible-role-nginx/blob/main/CONTRIBUTING.md) document
- [x] I have added Molecule tests that prove my fix is effective or that my feature works
- [x] I have checked that any relevant Molecule tests pass after adding my changes
- [x] I have updated any relevant documentation (`defaults/main/*.yml`, `README.md` and `CHANGELOG.md`)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
 # Changelog
 
-## 0.21.4 (Unreleased)
+## 0.22.0 (Unreleased)
+
+BREAKING CHANGES:
+
+* The `nginx_state` variable has been replaced with `nginx_setup` and instead of using `present`, `absent`, `latest` you should now use `install`, `uninstall` and `update`.
```
Comment (aknot242) [inline CHANGELOG.md:7]: May be a picky point, but would `upgrade` be a little more accurate than `update`?

File: defaults/main/main.yml [hunk 2]
```diff
@@ -27,12 +19,15 @@ nginx_type: opensource
...
-# Can be used with `nginx_version` to fine tune control on which version of NGINX is installed/used on each playbook execution.
-# Using 'present' will install the latest version (or 'nginx_version') of NGINX on a fresh install.
-# Using 'latest' will upgrade NGINX to the latest version (that matches your 'nginx_version') of NGINX on every playbook execution.
-# Using 'absent' will remove NGINX from your system.
-# Default is present.
-nginx_state: present
+# Can be used with `nginx_version` to fine tune con
+# Start NGINX service.
+# Default is true.
+nginx_start: truetrol on which version of NGINX is installed/used on each playbook execution.
```
Comment (aknot242) [inline defaults/main/main.yml:25]: Did `truetrol` come from part of line 22?

File: tasks/modules/install-modules.yml [hunk 3]
```diff
@@ -7,7 +7,7 @@
     - '"geoip" in nginx_modules'
     - nginx_install_epel_release | bool
 
-- name: Install NGINX modules
+- name: "Setup NGINX modules"
```
Comment (aknot242) [inline tasks/modules/install-modules.yml:10]: Why is just this task name surrounded with double quotes?

**Review by aknot242 — CHANGES_REQUESTED** [review]

File: defaults/main/main.yml (hunk 2 above)
Comment (alessfg) [reply defaults/main/main.yml]: Yep, might be what was causing tests to fail.

File: tasks/modules/install-modules.yml (hunk 3 above)
Comment (alessfg) [reply tasks/modules/install-modules.yml:10]: Good question, no real answer 😅

**Review by aknot242 — APPROVED** [review]

**Pull request merged**